### PR TITLE
dumpling: compatibility for `--no-locks`

### LINKED ISSUE #956: --no-locks not working in v2.0.0-rc
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

   see [askTUG](https://asktug.com/t/topic/36846/43).

   it should be compatible with `–consistency none`.

2. What did you expect to see?

3. What did you see instead?

4. Versions of the cluster

    - DM version (run `dmctl -V` or `dm-worker -V` or `dm-master -V`):

        ```
       v2.0.0-rc
        ```

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
close #956 

### What is changed and how it works?
replace `--no-locks` in `parseExtraArgs`

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch
 - Need to be included in the release note

### CODE REVIEW

File: dumpling/util.go [hunk 1]
```diff
@@ -58,8 +58,22 @@ func parseExtraArgs(logger *log.Logger, dumpCfg *export.Config, args []string) e
...
+		newargs         []string
 	)
 
+	// compatibility for `--no-locks`
+	for i, arg := range args {
+		if arg == "--no-locks" {
+			logger.Warn("`--no-locks` is replaced by `--consistency none` since v2.0.0")
+			newargs = args[:i]
+			newargs = append(newargs, "--consistency", "none")
+			newargs = append(newargs, args[i+1:]...)
```
Comment (csuzhangxc) [inline dumpling/util.go:70]: `break` here?

File: dumpling/util.go [hunk 2]
```diff
@@ -58,8 +58,22 @@ func parseExtraArgs(logger *log.Logger, dumpCfg *export.Config, args []string) e
...
 		filters         []string
+		newargs         []string
 	)
 
+	// compatibility for `--no-locks`
+	for i, arg := range args {
+		if arg == "--no-locks" {
+			logger.Warn("`--no-locks` is replaced by `--consistency none` since v2.0.0")
+			newargs = args[:i]
+			newargs = append(newargs, "--consistency", "none")
```
Comment (csuzhangxc) [inline dumpling/util.go:69]: if also specify another `--consistency`, can report an error in `dumplingFlagSet.Parse(args)`?

**Review by csuzhangxc — COMMENTED** [review]
rest LGTM

**Review by lichunzhu — APPROVED** [review]
LGTM

**Review by csuzhangxc — APPROVED** [review]
LGTM

**Pull request merged**